build/webpack/webpack.datascience-ui.config.builder.js

```diff
@@ -207,29 +207,44 @@ function buildConfiguration(bundle) {
                 {
                     test: /\.tsx?$/,
                     use: [
-                        {
-                            loader: 'thread-loader',
-                            options: {
-                                // there should be 1 cpu for the fork-ts-checker-webpack-plugin
-                                workers: require('os').cpus().length - 1,
-                                workerNodeArgs: ['--max-old-space-size=9096'],
-                                poolTimeout: isProdBuild ? 1000 : Infinity // set this to Infinity in watch mode - see https://github.com/webpack-contrib/thread-loader
-                            }
-                        },
-                        {
-                            loader: 'ts-loader',
-                            options: {
-                                happyPackMode: true, // IMPORTANT! use happyPackMode mode to speed-up compilation and reduce errors reported to webpack
-                                configFile: configFileName,
-                                // Faster (turn on only on CI, for dev we don't need this).
-                                transpileOnly: true,
-                                silent: true,
-                                compilerOptions: {
-                                    skipLibCheck: true
-                                },
-                                reportFiles: ['src/webviews/webview-side/**/*.{ts,tsx}']
-                            }
-                        }
+                        ...(fasterCompiler
```

Comment:
fasterCompiler is an existing variable we use, in the linter job we always compile slowly by checking types and the like.
When running tests, etc, we ignore all of that and compile as fast as possible (as we don't care about type checking since its done in a separate job)